src/content/app/species-selector/views/species-selector-results-view/SpeciesSelectorResultsView.tsx

```diff
@@ -73,19 +70,21 @@ const Content = (props: { onClose: () => void }) => {
     for (const addedGenome of genomes) {
       trackAddedGenome(addedGenome);
     }
+
+    navigate('/species-selector');
   };
 
-  return modalView === 'species-search' ? (
+  return searchParams.has('query') ? (
     <GenomeSelectorBySearchQuery
-      query={query}
+      query={searchParams.get('query') || ''}
       onSpeciesAdd={onSpeciesAdd}
       onClose={props.onClose}
     />
-  ) : selectedPopularSpecies ? (
+  ) : searchParams.has('species_taxonomy_id') && selectedPopularSpecies ? (
     <GenomeSelectorBySpeciesTaxonomyId
       onSpeciesAdd={onSpeciesAdd}
-      speciesTaxonomyId={selectedPopularSpecies.species_taxonomy_id}
-      speciesImageUrl={selectedPopularSpecies.image}
+      speciesTaxonomyId={selectedPopularSpecies?.species_taxonomy_id}
+      speciesImageUrl={selectedPopularSpecies?.image}
     />
   ) : null; // this last option should never happen
```

Comment:
We should probably be a bit smarter with the handling of the query parameters. Do we really want to show a mostly empty screen if the query parameter is neither `query` nor `species_taxonomy_id`; or do we want to rewrite the url such that we push the user to the main species selector screen?

Reply:
For now I am redirecting to `/species-selector` but we might need to check if this is alright with Andrea?

Reply:
✅